test/cpp/ring_buffer_test.cpp

```
@@ -0,0 +1,234 @@
+
+// ── SPSC stress: producer/consumer threads running concurrently ───────────────
+//
+// The producer writes 1-sample frames and the consumer reads them. After the
+// producer finishes, we drain remaining samples and verify that every value
+// written was eventually read in order (no corruption, no reordering).
+//
+// This exercises the release/acquire ordering on writeIndex/readIndex under
+// genuine concurrent access — a sanitizer run (TSAN) would catch races here.
+void testSpscStress() {
+    // Use a small ring so wrap-around happens frequently.
+    constexpr size_t kRingSize = 64;
```

Clang (14.0.6)

[warning] variable 'kRingSize' is not initialized

(cppcoreguidelines-init-variables)